site/content/entry/yuzu-progress-report-2018-p1-2/index.md

```diff
@@ -0,0 +1,237 @@
+This also applies to framebuffers because in some cases, games can use them as textures.
+If this caching wasn't existent, we'd essentially have to upload every texture to OpenGL memory (host GPU space) on every draw, copy the final framebuffer back to emulated Switch memory (CPU space) so that the game could potentially use it for framebuffer.
+
+{{< imgs
+    "./SMO_astc.png|Testing Super Mario Odyssey"
+    "./SMO.png|Inverted render"
+		"./SMO_2.png|Inverted render"
+>}}
+
+{{< imgs
+		"./SMO_debug.png|Debugging"
+		"./SMO_work.png| Proper render"
+		"./SMO_2_work.png| Proper render"
+>}}
+
+We had a couple of fixes to our rasterizer cache mainly because it was based on Citra's rasterizer cache.
+Because the compressed texture formats that the 3DS uses are not commonly supported by desktop GPUs, Citra's rasterizer cache decodes any compressed textures before uploading and caching them.
+But since the Switch GPU supports many of the same texture formats that desktop OpenGL does, we sped things up by uploading them without decoding them.
+At the start, this support was just hacked into Citra's cache, which didn't end up working correctly all the time.
+One example was the squid texture in "Splatoon 2", which was identified and fixed by Subv.
+We also implemented texture color swizzling, a way to swap color components, which is used by "Splatoon 2" to swap colors for intro background texture.
+So, before we implemented this the colors were all wrong.
+We fixed YUV2 video playback in our shader support, which fixed the previously busted "Sonic Mania" intro.
+We had to implement a field parameter called component masks in `TEX/TEXS` shader instructions, to properly decode `YUV2` videos.
+We also implemented texture wrap modes, which gives the ability to specify whether a texture on a triangle is mirrored or repeating etc.
+
+There are lots of other things already implemented and many more that have to be implemented.
+As the Switch's GPU is quite advanced, we have dozens of texture formats, vertex formats, lots of registers, different configuration modes, shader instructions that need to be implemented.
+
+{{< imgs
+    "./onepiece.png|One Piece - Then"
+		"./onepiece_work.png|One Piece - Now"
+    "./steam_world.png|Steam World Dig 2"
+>}}
+
+Apart from the above mentioned stuff, there have been many more modifications made to yuzu's GPU HLE (high level emulation).
+These bug fixes or modifications were done on a per-game experimentation basis and as we progress further we will continue to fix our implementations and improve the accuracy of emulation in yuzu.
+All of the progress we have made until now is thanks to the efforts of yuzu's contributors and the good people from Switch hacking communities.
+
+## Virtual File System (VFS) by [DarkLordZach](https://github.com/DarkLordZach) ([here](https://github.com/yuzu-emu/yuzu/pull/679))
```

Comment:
Are you sure this is the correct PR linked?

Reply:
Fixed by #34.